policy-engine/src/config/settings.rs

```diff
@@ -94,18 +95,33 @@ impl AppSettings {
             bail!("main and status service configured with the same address and port");
         }
 
+        // Deprecates options
+        if self.upstream.to_string() != hyper::Uri::default().to_string() {
+            warn!("the 'upstream' setting is deprecated and will eventually be removed.");
+        }
+
         Ok(self)
     }
-}
 
-fn default_openshift_plugins() -> Vec<BoxedPlugin> {
-    // TODO(lucab): drop this as soon as all config-maps are in place (prod & staging).
-    use cincinnati::plugins::internal::channel_filter::ChannelFilterPlugin;
-    use cincinnati::plugins::internal::metadata_fetch_quay::DEFAULT_QUAY_LABEL_FILTER;
-    use cincinnati::plugins::prelude::*;
-
-    new_plugins!(InternalPluginWrapper(ChannelFilterPlugin {
-        key_prefix: String::from(DEFAULT_QUAY_LABEL_FILTER),
-        key_suffix: String::from("release.channels"),
-    }))
+    fn default_openshift_policies(&self) -> Fallible<Vec<Box<dyn PluginSettings>>> {
```

Comment:
While reading this function, at first I didn't realize why the return type here needed to be Fallible.
To make it less surprising, it would be nice to move out the `?` which is currently hidden inside `plugin_config!()`.